src/field_theory/krull_topology.lean

```diff
@@ -287,3 +287,70 @@ begin
 end
 
 end totally_disconnected
+
+
+section finite_stuff
+
+variables {K E L : Type*} [field K] [field E] [algebra K E] [field L] [algebra K L]
+
+/-- A technical finiteness result. -/
+noncomputable def fintype.subtype_prod {E : Type*} {X : set E} (hX : X.finite) {L : Type*}
+  (F : E → multiset L) : fintype (Π x : X, {l : L // l ∈ F x}) :=
+by { classical, letI : fintype X := set.finite.fintype hX, exact pi.fintype}
+
+variable (K)
+
+/-- Function taking `e ∈ E` to the multiset of roots of its minimal polynomial in `L` -/
+noncomputable def roots_of_min_poly_multiset : E → multiset L :=
+λ e, ((minpoly K e).map (algebra_map K L)).roots
+
+variable [finite_dimensional K E]
+
+lemma minpoly.ne_zero' (e : E) : minpoly K e ≠ 0 :=
+minpoly.ne_zero $ is_integral_of_noetherian (is_noetherian.iff_fg.2 infer_instance) _
+
+variable (E)
+
+/-- The set of basis elements of `E` as a `K`-vector space -/
+def basis_set: set E :=
```

Comment:
It seems like you are not in a namespace. I think this name needs to be much more specific.